install/kubernetes/helm/subcharts/pilot/templates/deployment.yaml

```diff
@@ -92,6 +92,8 @@ spec:
               fieldRef:
                 apiVersion: v1
                 fieldPath: metadata.namespace
+          - name: ENABLE_INGRESS_GATEWAY_AGENT
+            value: "{{ .Values.ingressGatewayAgent.enabled }}"
```

Comment:
How is this any better than providing an API driven way for end users to enable/disable SDS? This is simply going to add more magic to helm and no one will know how to enable SDS for their gateways or sidecars in future.

Reply:
@costinm 
Basically, we need to tell Pilot whether Ingress gateway agent is enabled, to generate the right Envoy config for the gateway. We can use either helm config or configmap, and we are fine either way. Costin, what do you think?

Reply:
This should not be a user visible feature - we can add it to the mesh config, but I think it's better to stop polluting MeshConfig and API with features that are 1/2 baked or hurried up. 

Once we have some user feedback - we can figure out the right API.

Hacky APIs are far riskier than hacky environment variables.

Reply:
I would very much prefer for this to not be a pilot option - but passed as node metadata from ingress.

There is no need to make it a global - and keeping it as metadata will allow both gradual adoption and more flexibility.

Reply:
I'm not sure I understand "passed as node metadata from ingress". Does Envoy xDS request allow this metadata to be conveyed?

Reply:
When envoy starts, any environment variable starting with ISTIO_META is converted to node metadata, and sent in the request to pilot. 

The 'node' parameter passed to all methods has a map.

For sidecar - it includes labels and annotations of the pod, version of envoy, etc. 

In general using the metadata allows gradual deployment of a feature - you add annotations on 
a pod or deploy an ingress with the extra env, and it can be tested without changing a global setting and risking the entire cluster.

Reply:
Added an environment variable ISTIO_META_INGRESS_SDS for ingress gateway, that will be populated into node metadata as INGRESS_SDS: "true". Pilot will check this metadata and generate sds config for ingress gateway.